internal/core/software_renderer.rs

```diff
@@ -121,7 +121,7 @@ pub struct SoftwareRenderer {
     /// This is the area which was dirty on the previous frame.
     /// Only used if repaint_buffer_type == RepaintBufferType::SwappedBuffers
     prev_frame_dirty: Cell<DirtyRegion>,
-    window: Weak<dyn crate::window::WindowAdapter>,
+    window: RefCell<Option<Weak<dyn crate::window::WindowAdapter>>>,
```

Comment:
OnceCell ?
It is a bug to set a different window, right?

Reply:
I can make that change for the software renderer, if you prefer.